lib/samlp.js

```diff
@@ -32,10 +49,54 @@ function getSamlRequest(samlRequest, callback) {
           return callback(new Error('Invalid SAML Request'));
         }
 
-        callback(null, xml);
+        if (!options.signedRequest){
+          return callback(null, xml);
+        }
+        
+        // Http-Redirect binding can have the signature in query instead of in the AuthnRequest
+        // We pass the signature and algorithm if found.
+        if (options.signature){
+          if (!options.sigAlg){
+            return callback(new Error('Signature Algorithm is missing'));
+          }
+          
+          if (!algorithmSignatures[options.sigAlg]){
+            return callback(new Error('Invalid signature algorithm. Supported algorithms are http://www.w3.org/2001/04/xmldsig-more#rsa-sha1 and http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'));
+          }
+
+          if (!options.signingCert){
+            return callback(new Error('Make sure you configure the signing certificate to be able to validate the SAML Request'));
+          }
+
+          var signedParams; 
+          
+          if (options.relayState){
+            signedParams = {
+              SAMLRequest: samlRequest,
+              RelayState: options.relayState,
+              SigAlg: options.sigAlg
+            };
+          } else {
+            signedParams = {
+              SAMLRequest: samlRequest,
+              SigAlg: options.sigAlg
+            };
+          } 
+
+          try{
+            // Compare signatures
+            return verifySignature(options.signature, querystring.stringify(signedParams), certToPEM(options.signingCert), algorithmSignatures[options.sigAlg]) ?
+              callback(null, xml) :
+              callback(new Error('Signature check errors: The signature provided (' + options.signature + ') does not match the one calculated'));
+          } catch(e){
+            return callback(e);
+          }
+        }
+
+        validateXmlSignature(xml, { cert: options.signingCert, thumbprints: options.thumbprints }, callback);
```

Comment:
What is the difference between `verifySignature` (only for Redirect binding?) and `validateXmlSignature` (only for Post binding?)
Please add a comment and match both names (both should be `validate...` or `verify...`)